controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -135,9 +142,71 @@ func (r *StorageConsumerReconciler) initReconciler(request reconcile.Request) {
 
 func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error) {
 
+	secretList := &corev1.SecretList{}
+	if err := r.List(
+		r.ctx,
+		secretList,
+		client.MatchingFields{ownerIndexName: string(r.storageConsumer.UID)},
+		client.InNamespace(r.storageConsumer.Namespace),
+	); err != nil {
+		return reconcile.Result{}, err
+	} else if len(secretList.Items) > 1 {
+		return reconcile.Result{}, fmt.Errorf("expected 1 secret containing onboarding token for %s but found %d", r.storageConsumer.Name, len(secretList.Items))
+	}
```

Comment:
Since we already know the secret name why not get the secret directly instead of listing it?

Reply:
we don't want to tie ourselves with the name and so going via ownerrefs.

Reply:
done.